juriscraper/lib/network_utils.py

```diff
@@ -283,3 +285,58 @@ def get_attachments(
             self._map_docket_entry_document(attachment)
             for attachment in raw_attachments
         ]
+
+    def download_pdf(
+        self, document_id: str, page_count: int, document_url: str
+    ):
+        """
+        Requests generation and download of a merged PDF from the remote API.
+
+        This method sends a request to merge a document, polls the status
+        of the operation until it is completed, and then retrieves the PDF file
+        from the API.
+
+        :param document_id: identifier of the document in the ACMS system.
+        :param page_count: number of pages in the document.
+        :param document_url: The URL where the source document can be accessed.
+        :return: The HTTP response containing the merged PDF file.
+        """
+        # API endpoints for merge and retrieval
+        merge_pdf_path = f"{self.base_url}{self.MERGE_PDF_ENDPOINT}"
+        get_pdf_path = f"{self.base_url}{self.GET_PDF_ENDPOINT}"
+
+        # Request body specifying the document to merge
+        body = {
+            "header": True,
+            "mergeScope": "External",
+            "pagination": False,
+            "docketEntryDocuments": [
+                {
+                    "acms_docketdocumentdetailsid": document_id,
+                    "acms_documenturl": document_url,
+                    "acms_pagecount": page_count,
+                }
+            ],
+        }
+        # Step 1: Request PDF merge operation
+        response = self.session.post(merge_pdf_path, json=body)
+        response.raise_for_status()
+        data = response.json()
+
+        # Extract URL to check merge job status
+        status_url = data["statusQueryGetUri"]
+
+        # Step 2: Poll until the merge job is completed
+        while True:
+            response = self.session.get(status_url)
+            data = response.json()
+            status = data["runtimeStatus"].lower()
+            if status != "completed":
+                # Job still running, keep polling
+                continue
+            pdf_id = data["output"]
+            break
+
+        # Step 3: Retrieve the merged PDF by its file ID
+        response = self.session.get(f"{get_pdf_path}/?fileGuid={pdf_id}")
```

Comment:
We have that: `MERGE_PDF_ENDPOINT = "MergePDFFiles/"`


When we append the fileGuid parameter, we end up with a double slash `//`.

Maybe we can standardize the endpoints by removing trailing slashes:

```
MERGE_PDF_ENDPOINT = "MergePDFFiles"
GET_PDF_ENDPOINT = "GetMergedFile"
```

Also, can we pass `fileGuid` as a `params` argument instead of using an f-string?

Reply:
Also it's a good idea to add a `timeout` to this `get` request.

Reply:
> When we append the fileGuid parameter, we end up with a double slash //.

Thanks for pointing that out. I'll refine the code.